### ISSUE TITLE
Integrate alphabets to support universal transliteration logic

### ISSUE THREAD

**Issue by sergeyglazyrindev** [aaronland/go-ucd-username#5]
Hello Aaron.
I've checked your package. It's amazing.
But I am getting "cyrillic letter" instead of rune. It's ok, I understand how it works, but I've realized, it would be great to add ability to extend this package with different alphabets, like: for example, if UCDName.Name contains in the beginning: cyrillic, then we use cyrillic alphabet to transliterate rune into english.
And we can support alphabets as map[string]map[string]string, and for example we may initialize cyrillic alphabet in this map as :
`
"cyrillic": map[string]string{"а": "a", "б": "b"}, etc
`

**Comment by sergeyglazyrindev** [comment]
I can provide small POF.

**Comment by thisisaaronland** [comment]
Interesting. That sounds like it would be a useful `go-username` package in its own right, something that "sat on top" of this one.

That way all the logic around what (and whether) to transliterate is kept in one place and this package only concerns itself from mapping individual runes to their entry in the UCD file.

**Comment by sergeyglazyrindev** [comment]
Yes, I'd like to implement it as AlphabetRegistry, for example, we may define some default alphabets, our own version of transliteration, for example:
```
type AlphabetRegistry struct {
   alphabets map[string]map[string]string
}
func (ar *AlphabetRegistry) RegisterAlphabet(encoding string, alphabet map[string]string) {
   ar.alphabets[encoding] = alphabet
}
func (ar *AlphabetRegistry) GetAlphabet(encoding string) (map[string]string, error){
   alphabet, ok := ar.alphabets[encoding]
   if !ok {
     return nil, fmt.Errorf("no alphabet %s registered", encoding)
   }
     return alphabet, nil
}
```
in the UCD, when we determined what this rune is about, for example: cyrillic capital letter: , we split it by " " and then check if this alphabet is registered, and try to get the transliteration for the rune, if successful we use it.
This way we can give this package its own life, ability for developers to extend alphabets for their own needs.
I am busy few days but a bit later I'll implement POF.
